xcell/tests/test_mapper_dummy.py

```diff
@@ -30,26 +60,33 @@ def test_get_mask():
 
 
 def test_get_cl():
-    m = get_mapper()
-    c = get_config()
-    ls = 120
-    cl = 1. / (ls+c['l0'])**c['alpha']
-    rdev = np.fabs(cl / m.get_cl(ls) - 1)
-    assert np.max(rdev) < 1E-5
+    for dtype in ['galaxy_density', 'galaxy_shear', 'cmb_convergence']:
+        m = get_mapper(dtype)
+        cl_m = m.get_cl()
+
+        cl = get_cl(dtype)
+        rdev = np.fabs(cl[cl != 0] / cl_m[cl_m != 0] - 1)
+        assert np.max(rdev) < 1E-5
 
 
 def test_get_signal_map():
     m = get_mapper()
-    m.seed = 0
     d = m.get_signal_map()
-    d0 = [Table.read(
-        'xcell/tests/data/dummy_signal_s0.fits')['signal_map'].data]
+    d0 = [hp.read_map('xcell/tests/data/dummy_signal_s0.fits')]
 
+    # galaxy_density
     assert len(d) == 1
-    assert len(d0) == 1
-    d = d[0]
-    d0 = d0[0]
-    rdev = np.fabs(d / d0 - 1)
+    rdev = np.fabs(d[0] / d0[0] - 1)
+    assert np.max(rdev) < 1E-5
+    # galaxy_shear
+    m = get_mapper('galaxy_shear')
+    d = m.get_signal_map()
+    assert len(d) == 2
+    d0 = [hp.read_map('xcell/tests/data/dummy_signal_s2_0.fits'),
+          hp.read_map('xcell/tests/data/dummy_signal_s2_1.fits')]
+    rdev = np.fabs(d[0][d[0] != 0] / d0[0][d0[0] != 0] - 1)
```

Comment:
this has a risk of dividing by numbers that may be close to zero, leading to numerical inaccuracies. For this, I usually compare the standard deviation of the difference between both maps with the standard deviation of either of them, and make sure the former is several orders of magnitude smaller than the latter.

Reply:
Done